source/extensions/filters/http/common/aws/region_provider_impl.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include "common/common/logger.h"
+
+#include "extensions/filters/http/common/aws/region_provider.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Common {
+namespace Aws {
+
+class EnvironmentRegionProvider : public RegionProvider, public Logger::Loggable<Logger::Id::aws> {
+public:
+  absl::optional<std::string> getRegion() override;
+};
+
+class StaticRegionProvider : public RegionProvider {
```

Comment:
Sorry, use-case wise, can you explain more about the difference or the need of this Static vs Environment `RegionProvider`?

Reply:
Yes, please check the closed PR we are trying to break into smaller pieces here:

https://github.com/envoyproxy/envoy/pull/5546/files#diff-18af3435579912d71a19beaecd97d832R42

Reply:
I mean, I think to provide inline comments here probably helpful?

Reply:
Added documentation at the class level.